src/dftbp/timedep/linrespcommon.F90

```diff
@@ -758,119 +776,939 @@ subroutine actionAminusB(spin, wij, win, nocc_ud, nvir_ud, nxoo_ud, nxvv_ud, nxo
+          qv(:,jas) = qv(:,jas) + qij * vin(jbs) * sqrOccIA(jbs)
+        end do
+      end do
+
+      ! Attention: Necessary to diff. between nmat/nxov_rd and all possible exc
+      do jas = 1, sum(nxov_ud)
+        otmp(:) = qv(:,jas)
+        call hemv(qv(:,jas), lrGamma, otmp)
+      end do
+
+      do ias = 1, nmat
+        ii = getIA(win(ias), 1)
+        aa = getIA(win(ias), 2)
+        ss = getIA(win(ias), 3)
+        do jj = 1, nocc_ud(ss)
+          ijs = iaTrans(ii, jj, ss)
+          jas = iaTrans(jj, aa, ss)
+          qij(:) = transChrg%qTransIJ(ijs, env, denseDesc, ovrXev, grndEigVecs, getIJ)
+          vout(ias) = vout(ias) - cExchange * sqrOccIA(ias) * dot_product(qij, qv(:, jas))
+        end do
+      end do
+
+      qv(:,:) = 0.0_dp
+      do jbs = 1, nmat
+        jj = getIA(win(jbs), 1)
+        bb = getIA(win(jbs), 2)
+        ss = getIA(win(jbs), 3)
+        !! Here, index abs is different for a,b and b,a.
+        do aa = nocc_ud(ss) + 1, norb
+          jas = iaTrans(jj, aa, ss)
+          qij(:) = transChrg%qTransIA(jas, env, denseDesc, ovrXev, grndEigVecs, getIA, win)
+          abs = (bb - nocc_ud(ss) - 1) * nvir_ud(ss) + aa - nocc_ud(ss)
+          if (ss==2) then
+            abs = abs + nvir_ud(1)**2
+          end if
+          !! qv is not symmetric in a and b
+          qv(:,abs) = qv(:,abs) + qij * vin(jbs) * sqrOccIA(jbs)
+        end do
+      end do
+
+      do abs = 1, nxvv_a
+        otmp(:) = qv(:,abs)
+        call hemv(qv(:,abs), lrGamma, otmp)
+      end do
+
+      do ias = 1, nmat
+        ii = getIA(win(ias), 1)
+        aa = getIA(win(ias), 2)
+        ss = getIA(win(ias), 3)
+        do bb = nocc_ud(ss) + 1, norb
+          ibs = iaTrans(ii, bb, ss)
+          qij(:) = transChrg%qTransIA(ibs, env, denseDesc, ovrXev, grndEigVecs, getIA, win)
+          abs = (bb - nocc_ud(ss) - 1) * nvir_ud(ss) + aa - nocc_ud(ss)
+          if (ss==2) then
+            abs = abs + nvir_ud(1)**2
+          end if
+          vout(ias) = vout(ias) + cExchange * sqrOccIA(ias) * dot_product(qij, qv(:, abs))
+       end do
+      end do
+    endif
+
+    ! orb. energy difference diagonal contribution
+    vout(:) = vout + wij * vin
+    
+  end subroutine actionAminusB
+
+  !> Generates initial matrices M+ and M- for the RPA algorithm by Stratmann
+  !> (JCP 109 8218 (1998).
+  !> M+/- = (A+/-B)_ias,jbt (spin polarized) (A+/-B)^{S/T}_ia,jb (closed shell)
+  !> Here ias,jbt <= initDim
+  !> Also computed is v+/- = (A+/-B)_ias,jbt with ias <= nMat, jbt <= initDim
+  !> Note: Routine not set up to handle onsite corrections.
+  !> Note: Not yet OpenMP parallelized
+  subroutine initialSubSpaceMatrixApmB(transChrg, initDim, wIJ, sym, win, nmatup, env, denseDesc,&
+      & sTimesGrndEigVecs, grndEigVecs, occNr, sqrOccIA, getIA, getIJ, getAB, iaTrans, gamma,&
+      & lrGamma, species0, spinW, tSpin, tRangeSep, vP, vM, mP, mM)
+
+    !> machinery for transition charges between single particle levels
```

Comment:
See https://dftbplus-develguide.readthedocs.io/en/latest/fortranstyle.html#comments (capitalization).
Run [capitalize_ford_comments](https://github.com/dftbplus/dftbplus/blob/main/utils/srcmanip/capitalize_ford_comments), to comply with the style guide.